test/module.test.js

```diff
@@ -101,22 +108,108 @@ describe('ssr', () => {
   })
 })
 
-describe('generate', () => {
-  let nuxt
+describe('custom options', () => {
+  beforeAll(async () => {
+    nuxt = await setupNuxt({
+      ...config,
+      sitemap: {
+        path: 'my-sitemap.xml',
+        generate: true,
+        hostname
+      }
+    })
+  })
+
+  afterAll(async () => {
+    if (nuxt) {
+      await nuxt.close()
+    }
+  })
+
+  test('includes all static routes', async () => {
+    const xml = await get('/my-sitemap.xml')
+
+    // static routes
+    expect(xml).toContain(`<loc>${hostname}</loc>`)
+    expect(xml).toContain(`<loc>${hostname}sub</loc>`)
+    expect(xml).toContain(`<loc>${hostname}sub/sub</loc>`)
+
+    // static child-routes
+    expect(xml).toContain(`<loc>${hostname}parent</loc>`)
+    expect(xml).toContain(`<loc>${hostname}parent/child</loc>`)
+    expect(xml).toContain(`<loc>${hostname}parent/child/subchild</loc>`)
+    expect(xml).not.toContain(`<loc>${hostname}parent/</loc>`)
+    expect(xml).not.toContain(`<loc>${hostname}parent/child/</loc>`)
+
+    // dynamic routes
+    expect(xml).toContain(`<loc>${hostname}child</loc>`)
+    expect(xml).not.toContain(`<loc>${hostname}child/1</loc>`)
+    expect(xml).not.toContain(`<loc>${hostname}child/2</loc>`)
+
+    // excluded routes
+    expect(xml).toContain(`<loc>${hostname}exclude</loc>`)
+
+    // filtered routes
+    expect(xml).toContain(`<loc>${hostname}filtered</loc>`)
+  })
+
+  test('custom sitemap path', async () => {
+    await expect(get('/sitemap.xml')).rejects.toMatchObject({
+      statusCode: 404
```

Comment:
I will not work. The "test/fixture/pages/_param" catch this route, so the statusCode = 200